src/boot/efi/efi-string.c

```diff
@@ -272,6 +274,453 @@ bool efi_fnmatch(const char16_t *pattern, const char16_t *haystack) {
+        [NOERR(EFI_BAD_BUFFER_SIZE)]      = "Bad buffer size",
+        [NOERR(EFI_BUFFER_TOO_SMALL)]     = "Buffer too small",
+        [NOERR(EFI_NOT_READY)]            = "Not ready",
+        [NOERR(EFI_DEVICE_ERROR)]         = "Device error",
+        [NOERR(EFI_WRITE_PROTECTED)]      = "Write protected",
+        [NOERR(EFI_OUT_OF_RESOURCES)]     = "Out of resources",
+        [NOERR(EFI_VOLUME_CORRUPTED)]     = "Volume corrupt",
+        [NOERR(EFI_VOLUME_FULL)]          = "Volume full",
+        [NOERR(EFI_NO_MEDIA)]             = "No media",
+        [NOERR(EFI_MEDIA_CHANGED)]        = "Media changed",
+        [NOERR(EFI_NOT_FOUND)]            = "Not found",
+        [NOERR(EFI_ACCESS_DENIED)]        = "Access denied",
+        [NOERR(EFI_NO_RESPONSE)]          = "No response",
+        [NOERR(EFI_NO_MAPPING)]           = "No mapping",
+        [NOERR(EFI_TIMEOUT)]              = "Time out",
+        [NOERR(EFI_NOT_STARTED)]          = "Not started",
+        [NOERR(EFI_ALREADY_STARTED)]      = "Already started",
+        [NOERR(EFI_ABORTED)]              = "Aborted",
+        [NOERR(EFI_ICMP_ERROR)]           = "ICMP error",
+        [NOERR(EFI_TFTP_ERROR)]           = "TFTP error",
+        [NOERR(EFI_PROTOCOL_ERROR)]       = "Protocol error",
+        [NOERR(EFI_INCOMPATIBLE_VERSION)] = "Incompatible version",
+        [NOERR(EFI_SECURITY_VIOLATION)]   = "Security violation",
+        [NOERR(EFI_CRC_ERROR)]            = "CRC error",
+        [NOERR(EFI_END_OF_MEDIA)]         = "End of media",
+        [29]                              = "Reserved (29)",
+        [30]                              = "Reserved (30)",
+        [NOERR(EFI_END_OF_FILE)]          = "End of file",
+        [NOERR(EFI_INVALID_LANGUAGE)]     = "Invalid language",
+        [NOERR(EFI_COMPROMISED_DATA)]     = "Compromised data",
+#  ifdef EFI_HTTP_ERROR
+        [NOERR(EFI_IP_ADDRESS_CONFLICT)]  = "IP address conflict",
+        [NOERR(EFI_HTTP_ERROR)]           = "HTTP error",
+#  endif
+#endif
+};
+
+static const char *process_format_specifier(
+                EFI_STATUS status,
+                const char *format,
+                va_list *ap,
+                char16_t tmp[static SCRATCH_BUF_SIZE],
+                bool *ret_pad_zero,
+                int *ret_precision,
+                int *ret_field_width,
+                const char **ret_str8,
+                const char16_t **ret_str16,
+                const wchar_t **ret_wstr) {
+
+        assert(ap);
+        assert(ret_pad_zero);
+        assert(ret_precision);
+        assert(ret_field_width);
+        assert(ret_str8);
+        assert(ret_str16);
+        assert(ret_wstr);
+
+        bool alternative_form = false, have_field_width = false, long_arg = false;
+        size_t field_length = sizeof(int);
+        int precision = -1, field_width = -1;
+
+        for (;; format++)
+                switch (*format) {
+                case '0':
+                        *ret_pad_zero = true;
```

Comment:
we have this coding style rule that on success we initialize all return params, and on failure none. afaics this might be set or might be left alone on success so this is conflicting with this rule.

You might be able to get away by simply dropping the "ret_" prefix, i.e. by stopping to claim that this is a return paremeter, but instead an input/output param.